### PR TITLE
th-81: Create order card component

### PR DESCRIPTION
**Pull request by mV22110702**

### CODE REVIEW

**Review by Druxight — DISMISSED** [review]

File: frontend/src/libs/components/card/card.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}: Properties) => (
+  <div className={style.container}>
+    <div className={style.card_layout}>
+      <div className={style.horizontal_bar}>
+        <div>
+          <PlainSvgIcon name={PlainSvgIconName.HORIZONTAL_BAR} />
+        </div>
+      </div>
+      <div className={style.header}>
+        <div className={style.header_image_container}>
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/card/card.tsx]: I believe we decided to use camelCase for classnames

File: frontend/src/libs/components/card/card.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          <img className={style.profileImage} src={profileURL} alt="header" />
+        </div>
+        <div className={style.header_info_container}>
+          <div className={style.header_title_container}>
+            <span className={getValidClassNames('text-md')}>
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/card/card.tsx]: do we need getValidClassNames here?

File: frontend/src/libs/components/card/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,157 @@
+$row-gap-sm: 0.5vmin;
+$row-gap-md: 1vmin;
+$row-gap-lg: 2vmin;
+$shadow-dark: 0 4px 4px #00000021;
+$grey-darker: #7e818f;
+$violet-shallow: #93a1ff;
+$blue-shallower: #6c91f9;
+$blue-shallow: #becee8;
+
+@import "../../../assets/css/styles.scss";
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/card/styles.module.scss]:
```suggestion
@import "src/assets/css/styles.scss";
```

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]

**Review by Druxight — DISMISSED** [review]

File: frontend/src/libs/components/components.ts [hunk 4]
```diff
@@ -13,6 +13,8 @@ export { Image } from './image/image.js';
 export { Input } from './input/input.js';
 export { Link } from './link/link.js';
 export { Modal } from './modal/modal.js';
+export { OrderCard } from './order-card/order-card.js';
+export { PlainSvgIcon } from './plain-svg-icon/plain-svg-icon.jsx';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/components.ts]: ?
```suggestion
export { PlainSvgIcon } from './plain-svg-icon/plain-svg-icon.js';
```

File: frontend/src/libs/components/order-card/order-card.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
+import { PlainSvgIconName } from '~/libs/enums/enums.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+
+import { PlainSvgIcon } from '../plain-svg-icon/plain-svg-icon.jsx';
+import style from './styles.module.scss';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-card/order-card.tsx]:
```suggestion
import styles from './styles.module.scss';
```

File: frontend/src/libs/components/plain-svg-icon/maps/red-circle.svg
Comment (kalinkaaaa14) [inline frontend/src/libs/components/plain-svg-icon/maps/red-circle.svg:1]: I don't like that these images will be stored inside libs/components/plain-svg-icon/maps, let's move it somewhere in assets/img? Or public/img

File: frontend/src/libs/components/plain-svg-icon/maps/maps.tsx [hunk 6]
```diff
@@ -0,0 +1 @@
+export { iconNameToPlainSvgMap } from './icon-name-to-plain-svg-map.jsx';
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/plain-svg-icon/maps/maps.tsx]:
```suggestion
export { iconNameToPlainSvgMap } from './icon-name-to-plain-svg-map.js';
```

File: frontend/src/libs/components/order-card/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,156 @@
+$row-gap-sm: 0.5vmin;
+$row-gap-md: 1vmin;
+$row-gap-lg: 2vmin;
+$shadow-dark: 0 4px 4px #00000021;
+$grey-darker: #7e818f;
+$violet-shallow: #93a1ff;
+$blue-shallower: #6c91f9;
+$blue-shallow: #becee8;
+
+@import "src/assets/css/styles.scss";
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-card/styles.module.scss]: I prefer imports at the top of files

File: frontend/src/libs/components/order-card/order-card.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+type Properties = {
+  driver: {
+    firstName: string;
+    lastName: string;
+    profileURL: string;
+  };
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-card/order-card.tsx]: maybe let's create separate type for it called for example DriverProfileInfo or something like this?

File: frontend/src/libs/components/order-card/order-card.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+            </div>
+            <span className={style.routeInfo}>{endLocation}</span>
+          </div>
+        </div>
+        <div className={style.distanceContainer}>
+          <div className={style.distanceIcon}>
+            <PlainSvgIcon name={PlainSvgIconName.MAP} />
+          </div>
+          <span className={getValidClassNames(style.distanceInfo, 'text-md')}>
+            {distanceLeft} km{distanceLeft > 1 && 's'}, {timespanLeft}
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-card/order-card.tsx]: let's create a variable for calculation if there should be km or kms outside the jsx

**Review by kalinkaaaa14 — CHANGES_REQUESTED** [review]

**Comment by kalinkaaaa14** [comment]
don't forget to rerequest for a review, and remove labels (conflicts are resolved for ex)

**Review by PavloSukhinin — DISMISSED** [review]

File: frontend/src/libs/components/plain-svg-icon/maps/icon-name-to-plain-svg-map.tsx
Comment (kalinkaaaa14) [inline frontend/src/libs/components/plain-svg-icon/maps/icon-name-to-plain-svg-map.tsx:1]: filename - icon-name-to-plain-svg.map
map name - iconNameToPlainSvg

**Review by kalinkaaaa14 — APPROVED** [review]

**Review by PavloSukhinin — APPROVED** [review]

**Pull request merged**